README.md

```diff
@@ -2,9 +2,50 @@
 
-This package contains the Golang interface for using the Windows [Host Compute Service](https://techcommunity.microsoft.com/t5/containers/introducing-the-host-compute-service-hcs/ba-p/382332) (HCS) to launch and manage [Windows Containers](https://docs.microsoft.com/en-us/virtualization/windowscontainers/about/). It also contains other helpers and functions for managing Windows Containers such as the Golang interface for the Host Network Service (HNS).
+This package contains the Golang interface for using the Windows [Host Compute Service](https://techcommunity.microsoft.com/t5/containers/introducing-the-host-compute-service-hcs/ba-p/382332) (HCS) to launch and manage [Windows Containers](https://docs.microsoft.com/en-us/virtualization/windowscontainers/about/). It also contains other helpers and functions for managing Windows Containers such as the Golang interface for the Host Network Service (HNS), as well as code for the guest agent used to support running Linux Hyper-V containers.
 
-It is primarily used in the [Moby Project](https://github.com/moby/moby), but it can be freely used by other projects as well.
+It is primarily used in the [Moby](https://github.com/moby/moby) and [Containerd](https://github.com/containerd/containerd) projects, but it can be freely used by other projects as well.
+
+## Building
+
+While this repository can be used as a library of sorts to call the HCS apis, there are a couple binaries built out of the repository as well. The main ones being the Linux guest agent, and an implementation of the [runtime v2 containerd shim api](https://github.com/containerd/containerd/blob/master/runtime/v2/README.md).
+### Linux Hyper-V Container Guest Agent
+
+To build the Linux guest agent itself all that's needed is to set your GOOS to "Linux" and build out of ./cmd/gcs.
+```powershell
+C:\> $env:GOOS="linux"
+C:\> go build .\cmd\gcs\
+```
+
+or on a Linux machine
+```sh
+> go build ./cmd/gcs
+```
+
+If you want it to be packaged inside of a rootfs to boot with alongside all of the other tools then you'll need to provide a rootfs that it can be packaged inside of. An easy way is to export the rootfs of a container.
+
+```sh
+docker pull busybox
+docker run --name base_image_container busybox
+docker export base_image_container | gzip > base.tar.gz
+BASE=./base.tar.gz
+make all
+```
+
+If the build is successful, in the `./out` folder you should see:
+```sh
+> ls ./out/
+delta.tar.gz  initrd.img  rootfs.tar.gz
+```
+
+### Containerd Shim
+The containerd shim is very simple to build.
+```powershell
+C:\> go build .\cmd\containerd-shim-runhcs-v1
+```
+
+## OpenGCS
+The Linux guest agent used to be developed out of the https://github.com/Microsoft/opengcs repository, but all further development will take place here. The majority of the code lives in the [./internal/guest](./internal/guest/README.md) directory.
```

Comment:
I think maybe we should move the information on opengcs and how to build it into the readme in internal/guest and here just have a brief overview of what it is and a link to that readme for more info? What do you think?

Reply:
I kinda liked it here as if someone winds up on the repo and is looking for a quick "how do I build" it felt weird to mention how to build one half of the equation (the shim) and then follow a link to an internal README that tells you how to build the other half, which is just a command in the root of the directory anyways. 

I flip flopped on this multiple times when writing these up. I was going with the approach we originally came up with when speaking on this, but I'm not tied to it. What do you think? The internal/guest README is a bit barebones I'll admit.

Reply:
I think this is fine given that perspective